backend/components/ibm-watson-assistant/src/main/java/co/airy/core/ibm-watson-assistant-connector/MessageHandler.java

```diff
@@ -0,0 +1,104 @@
+package co.airy.core.ibm_watson_assistant_connector;
+
+import co.airy.avro.communication.DeliveryState;
+import co.airy.avro.communication.Message;
+import co.airy.core.ibm_watson_assistant.models.MessageSendResponse;
+import co.airy.log.AiryLoggerFactory;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.springframework.stereotype.Service;
+import org.slf4j.Logger;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.UUID;
+
+@Service
+public class MessageHandler {
+    private final ObjectMapper mapper = new ObjectMapper();
+    private static final Logger log = AiryLoggerFactory.getLogger(MessageHandler.class);
+
+    MessageHandler() {
+    }
+
+    public Message getMessage(Message contactMessage, MessageSendResponse response) throws Exception {
+        String content = getContent(contactMessage.getSource(), response);
+        if (content == null) {
+            throw new Exception("Unable to map IBM Watson Assistant reply to source response.");
+        }
+
+        return Message.newBuilder()
+                .setId(UUID.randomUUID().toString())
+                .setChannelId(contactMessage.getChannelId())
+                .setContent(content)
+                .setConversationId(contactMessage.getConversationId())
+                .setHeaders(Map.of())
+                .setDeliveryState(DeliveryState.PENDING)
+                .setSource(contactMessage.getSource())
+                .setSenderId("ibm-watson-assistant-bot")
+                .setSentAt(Instant.now().toEpochMilli())
+                .setIsFromContact(false)
+                .build();
+    }
+
+    public String getContent(String source, MessageSendResponse response) throws JsonProcessingException {
+
+        try {
+            final JsonNode responseNode = mapper.valueToTree(response);
+            String text = "";
+
+            for (JsonNode nestedNode : responseNode.path("output").path("generic")) {
+                text = nestedNode.get("text").textValue();
+            }
+
```

Comment:
in the response from the AI assistant, the text is nested inside of the `output.generic` array.